crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -784,10 +783,27 @@ impl PendingTransactions {
         let mut queue = Vec::with_capacity(self.len());
         // Add all transactions to the queue.
         for (address, account_txs) in &self.txs {
-            let current_account_nonce = state
-                .get_account_nonce(address)
-                .await
-                .wrap_err("failed to fetch account nonce for builder queue")?;
+            let current_account_nonce = match state.get_account_nonce(address).await {
```

Comment:
if this function returns an error, it's pretty severe (either we can't read from state, and the app will crash soon anyways, or we can't deserialize the storage value which would have caused an error earlier) so i think `expect()`ing here is fine

Reply:
`expect`ing is not a replacement for error handling. `expect`ing should only be used for when system invariants are so badly violated that you can't recover from them at all, which is not the case here.